tutorials/time-freq/plot_ssvep.py

```diff
@@ -0,0 +1,767 @@
+# You can see that the peaks at the stimulation frequencies (12 hz, 15 hz)
+# and their harmonics are visible in both plots.
+# Yet, the SNR spectrum shows them more prominently as peaks from a
+# noisy but more or less constant baseline of SNR = 1.
+# You can further see that the SNR processing removes any broad-band power
+# differences (such as the increased power in alpha band around 10 hz),
+# and also removes the 1/f decay in the PSD.
+#
+# Note, that while the SNR plot implies the possibility of values below 0
+# (mean minus sd) such values do not make sense.
+# Each SNR value is a ratio of positive PSD values, and the lowest possible PSD
+# value is 0 (negative Y-axis values in the upper panel only result from
+# plotting PSD on a log scale).
+# Hence SNR values must be positive and can minimally go towards 0.
+# You can nicely see this at 50hz with the artifact induced by notch
+# filtering the line noise:
+# The PSD spectrum shows a prominent negative peak and average SNR approaches
+# 0.
+#
+# Subsetting data
+# ---------------
+#
+# Our processing yielded a large array of many SNR values for each trial x
+# channel x frequency-bin of the PSD array.
+#
+# For statistical analysis we obviously need to define specific subsets of this
+# array. First of all, we are only interested in SNR at the stimulation
+# frequency, but we also want to restrict the analysis to a spatial ROI. The
+# most interesting questions, however, will probably rely on comparing SNR in
+# different trials.
+#
+# Since we here have a large SNR array with all conditions, we will have to
+# find the indices of trials, channels, etc.
+# Alternatively, one could subselect the trials already at the epoching step,
+# using MNE's event information, and process different epoch structures
+# individually.
+#
+# Let's have a look at the trials with 12 hz stimulation, for now.
+#
+
+# define stimulation frequency
+stim_freq = 12.
+
+###############################################################################
+# Get index for the stimulation frequency (12hz)
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# Ideally, there would be a bin with the stimulation frequency exactly in its
+# center. However, depending on your Spectral decomposition this is not
+# always the case. We will find the bin closest to it (this one should contain
+# our frequency tagged response.
+#
+
+# find index of frequency bin closest to stimulation frequency
+i_bin_12hz = np.argmin(abs(np.subtract(freqs, stim_freq)))
+# could be updated to support multiple frequencies
+
+# for later, we will already find the 15 hz bin and the 1st and 2nd harmonic
+# for both.
+i_bin_24hz = np.argmin(abs(np.subtract(freqs, 24)))
+i_bin_36hz = np.argmin(abs(np.subtract(freqs, 36)))
+i_bin_15hz = np.argmin(abs(np.subtract(freqs, 15)))
+i_bin_30hz = np.argmin(abs(np.subtract(freqs, 30)))
+i_bin_45hz = np.argmin(abs(np.subtract(freqs, 45)))
```

Comment:
you should use 

https://numpy.org/doc/stable/reference/generated/numpy.searchsorted.html

Reply:
I think it would work differently when there is not an exact frequency in `freqs`. If the frequencies are [..., 1.44, 1.66, 1.88, ...] and the stimulation frequency is 1.7 Hz, we would like to find the index of 1.66, not 1.88 as `searchsorted` would do.

Reply:
combine it with np.round?

Reply:
I am not sure how we could use rounding. We could round the stimulation frequency to the nearest whole multiple of the frequency step but this seems like an overly complicated way to look for the value we need. Do you see a simpler solution?